sanctions/apps/sanctions/models.py

```diff
@@ -0,0 +1,84 @@
+from simple_history.models import HistoricalRecords
+
+from django.core.validators import MinLengthValidator
+from django.db import models
+from django_extensions.db.models import TimeStampedModel
+from django.utils.translation import gettext_lazy as _
+
+class SanctionsCheckFailure(TimeStampedModel):
+    """ Record of SDN and ISN check failure. """
+    history = HistoricalRecords()
+    full_name = models.CharField(max_length=255)
+    username = models.CharField(max_length=255)
+    lms_user_id = models.IntegerField(null=True, db_index=True)
+    city = models.CharField(max_length=32, default='')
+    country = models.CharField(max_length=2)
+    sanctions_type = models.CharField(max_length=255)
+    system_identifier = models.CharField(max_length=255)
+    order_identifier = models.CharField(max_length=128, default=None, null=True)
```

Comment:
This would be the equivalent to an edX `order_number`, Commercetools creates an order before payment is processed, a cart is also created however a cart does not have billing address information.

Reply:
i have mixed feelings about the sanctions service knowing about what an order is (the number, type, price etc). this effectively declares sanctions service be directly for checkout related things... which, it could be, but should it? if this service's purpose is "to check a user against sanctions lists, and then carry out action on positive hits", this technically doesn't need to have anything to do with checkout, and in theory could happen at some other point during a user's journey (i know seth had been exploring other SDN options for the future that potentially occur outside of checkout) 

if we truly want this service to be independent and not be bounded to checkout, which i think we should strive for, the order details should probably be housed in a different field or table 🤔 i'm interested to hear more about your thoughts on table joins though

Reply:
i want to think about this a bit, because i don't think having these fields is the end of the world or anything, but want to think through to make sure we are making the right abstraction layers

Reply:
Sounds good, I added these without migrations for some discussion.
I thought SDN checks are done because we cannot sell anything to persons on this list, how would it not be related to checkout and done in another point during the user's journey?

`if we truly want this service to be independent and not be bounded to checkout`
I was not aware this is what we're shooting for, if that's the case then it makes sense not to have checkout related information

Reply:
Update: we'll use a `metadata` JSONfield like Chris recommended and remove any "order" named fields.